website/docs/r/network_interface.html.markdown

```diff
@@ -91,11 +92,13 @@ The `ip_configuration` block supports:
 
 * `load_balancer_inbound_nat_rules_ids` - (Optional) List of Load Balancer Inbound Nat Rules IDs involving this NIC
 
+* `primary` - (Optional) Is this the Primary Network Interface? If set to `true` this should be the first `ip_configuration` in the array.
```

Comment:
> If set to `true` this should be the first `ip_configuration` in the array

This feels a little bit unintuitive for the user - is it because the API requires it? Can't we just sort it for the user prior to sending the request to the API?